src/stats/favorite-players/likes.ts

```diff
@@ -0,0 +1,139 @@
+import { Table } from 'craftjs-plugin/database';
+import { UUID } from 'java.util';
+import { Bukkit, OfflinePlayer } from 'org.bukkit';
+import { Player } from 'org.bukkit.entity';
+import { errorMessage } from '../../chat/system';
+import { getTable } from '../../common/datas/database';
+import { isAdminAccount } from '../../common/helpers/player';
+import { getTranslator } from '../../common/localization/localization';
+import { displayLikeSuccess } from './messages';
+import { PlayerLikes } from './PlayerLikes';
+
+/*********************
+ * Exported functions
+ *********************/
+
+export function getTopLikeList(howMany: number): PlayerLikes[] {
+  refreshLikeList();
+  const sorted = getSortedLikeList();
+  return sorted.slice(0, howMany);
+}
+
+export function addLikeToUsername(sender: Player, username: string) {
+  const target = Bukkit.getOfflinePlayer(username);
+  addLike(target);
+  displayLikeSuccess(sender, target);
+}
+
+export function canLike(liker: Player, username: string) {
+  const target = Bukkit.getOfflinePlayer(username);
+  const tr = getTranslator(liker);
+
+  if (!isValidPlayer(target)) {
+    errorMessage(liker, tr('likes.player_not_found'));
+    return false;
+  }
+  if (isAdminAccount(target)) {
+    errorMessage(liker, tr('likes.cannot_like_admins'));
+    return false;
+  }
+  if (target == liker) {
+    errorMessage(liker, tr('likes.cannot_like_yourself'));
+    return false;
+  }
+
+  return true;
+}
+
+/************************
+ * Private functions
+ ************************/
+
+const likesDb: Table<string, number> = getTable('likes-table');
```

Comment:
Databases (backed by H2 MVStore) support UUIDs natively as keys and values. They *probably* take less space than UUID strings. If this doesn't work, update craftjs-plugin types.

Also, no need to postfix with -table